packages/angular/ssr/src/routes/ng-routes.ts

```diff
@@ -139,12 +167,71 @@ async function* traverseRoutesConfig(options: {
           compiler,
           parentInjector: injector,
           parentRoute: currentRoutePath,
+          serverConfigRouteTree,
+          invokeGetPrerenderPaths,
         });
       }
     }
   }
 }
 
+/**
+ * Handles SSG (Static Site Generation) routes by invoking `getPrerenderPaths` and yielding
+ * all parameterized paths.
+ *
+ * @param metadata - The metadata associated with the route tree node.
+ * @param parentInjector - The dependency injection container for the parent route.
+ * @param invokeGetPrerenderPaths - A flag indicating whether to invoke the `getPrerenderPaths` function.
+ * @returns An async iterable iterator that yields route tree node metadata for each SSG path.
+ */
+async function* handleSSGRoute(
+  metadata: ServerConfigRouteTreeNodeMetadata,
+  parentInjector: Injector,
+  invokeGetPrerenderPaths: boolean,
+): AsyncIterableIterator<RouteTreeNodeMetadata> {
+  if (metadata.renderMode !== RenderMode.SSG) {
+    return;
+  }
+
+  const { route: currentRoutePath, getPrerenderPaths, fallback, ...meta } = metadata;
+
+  if (invokeGetPrerenderPaths && currentRoutePath.includes(':')) {
+    if (!getPrerenderPaths) {
+      throw new Error(
+        `The Angular route '${currentRoutePath}' includes parameters, but 'getPrerenderPaths' is missing. ` +
+          `Please define 'getPrerenderPaths' in your server routing configuration.`,
+      );
+    }
+
+    const parameters = await runInInjectionContext(parentInjector, () => getPrerenderPaths());
```

Comment:
Question: How does `runInInjectionContext` handle async callbacks? `getPrerenderPaths()` might return a `Promise`. What happens for:

```typescript
function getPrerenderPaths() {
  await someAsyncWork();
  const service = inject(SomeService); // Can we inject after an `await`?
}
```

Reply:
The `inject` needs to happen before any async work.

Reply:
Is there any precedent for Angular APIs which are naturally async but run in an injection context? This feels like an easy tripping hazard.

I guess the workaround would be to keep a reference to the injector (or inject before the `await` like you suggested)?

```typescript
async function getPrerenderPaths() {
  const injector = inject(Injector);
  const result = await someAsyncWork();
  if (result === something) {
    injector.inject(SomethingElse);
  }
}
```

/cc @AndrewKushnir is this something we should be concerned about?

Reply:
Both guards and interceptors run in an injection context.